bittensor/config.py

```diff
@@ -223,38 +225,55 @@ def __init__(
             args=args, parser=parser_no_defaults, strict=strict
         )
 
-        ## Diff the params and params_no_defaults to get the is_set map
-        _config["__is_set"] = {
-            arg_key: True
-            for arg_key in [
-                k
-                for k, _ in filter(
-                    lambda kv: kv[1] != argparse.SUPPRESS,
-                    params_no_defaults.__dict__.items(),
-                )
-            ]
-        }
+        # TODO: Where to store defaults? __init__ defaults are not usable here because of circular imports
```

Comment:
@LogrisTheBard please review / comment on TODO

Reply:
I'm fine with storing the profile path default here but I don't think this method would use the env of profile.path or config.path when loading those respective files.

I'd suggest modifying the load_ methods to receive a dict of the config to use to find the file and to return a config that we can merge together several times for the full process.

Prior to looking up the generic config file we should have a config.path resulting from (in priority order) the args_no_defaults, the env vars, and then the defaults.

After we have the generic config and profile, then it merges again in a different order so as to preserve the defaults being lowest priority: args, envs, profile, config, default.